File: HW1/Rectangle.cpp
#include"Rectangle.h"

int Rectangle::numOfRect = 0;

Rectangle::Rectangle() {
	numOfRect++;
}
Rectangle::Rectangle(double width, double height) {
	setHeight(height);
	setWidth(width);
}

void Rectangle::setWidth(double newWidth) {
	width = newWidth;
}
void Rectangle::setHeight(double newHeight) {
	height = newHeight;
}

double Rectangle::getWidth() const {
	return width;
}
double Rectangle::getHeight() const {
	return height;
}

double Rectangle::getArea() const {
	return height * width;
}
double Rectangle::getPerimeter() const {
	return (width + height) * 2;
}
int Rectangle::getNumOfRect() {
	return numOfRect;
}

File: Python/Rectangle_Encapsulation/Rectangle.py
class Rectangle:
    # Construct, def is func
    # private: length, width(__)
   def __init__(self, width=1 , length=1):
       self.__width = width
       self.__length = length
    
    # self 是呼叫自己 private 時必須的
    # setFunc, __length is private, __width is private
   def setLength(self, length):
        self.__length = length

   def setWidth(self, width):
        self.__width = width

    # getFunc
   def getLength(self):
       return self.__length
   def getWidth(self):
       return self.__width

    # Calculation
   def getArea(self):
        return self.__length * self.__width
   def getPerimeter(self):
       return 2 * (self.__width + self.__length)
